File: lecture-recording-manager/ClientApp/src/app/shared/recording-output.ts
export class RecordingOutput {
    id: number;
    recordingId: number;
    jobType: string;
    jobConfiguration: string;
    status: RecordingStatus;
    processed: boolean;
    jobError: string;
    dateStarted: Date;
    dateFinished: Date;
}

export enum RecordingStatus {
    UPLOADED = 0,
    PROCESSING = 1,
    PROCESSED = 2,
    SCHEDULED = 3,
    ERROR = -1
}

File: lecture-recording-manager/Models/RecordingOutput.cs
using Newtonsoft.Json;
using System;

namespace LectureRecordingManager.Models
{
    public class RecordingOutput
    {
        public int Id { get; set; }

        public int RecordingId { get; set; }

        [JsonIgnore]
        public Recording Recording { get; set; }

        public string JobType { get; set; }

        public string JobConfiguration { get; set; }

        public RecordingStatus Status { get; set; }

        public bool Processed { get; set; }

        public string JobError { get; set; }

        public DateTimeOffset? DateStarted { get; set; }

        public DateTimeOffset? DateFinished { get; set; }
    }
}
